modules/db/src/main/java/org/mule/module/db/internal/config/domain/query/ParameterizedQueryTemplateFactoryBean.java

```diff
@@ -136,6 +137,14 @@ public boolean evaluate(Object object)
         return null;
     }
 
+    private Object checkNullValue(Object value)
```

Comment:
If you can define a NULL value on a template, then it makes sense to be able to set NULL on a db:input-param/db:inout-param.
I would only accept NULL to represent a null.
Also, note that this change must be properly documented on the migration guide.